octobot_evaluators/evaluator/realtime_evaluator.py

```diff
@@ -36,3 +37,26 @@ def get_symbol_candles(self, exchange_name: str, exchange_id: str, symbol: str,
         except ImportError:
             self.logger.error(f"Can't get candles manager: requires OctoBot-Trading package installed")
 
+    def _get_tentacle_registration_topic(self, all_symbols_by_crypto_currencies, time_frames, real_time_time_frames):
+        currencies, symbols, _ = super()._get_tentacle_registration_topic(all_symbols_by_crypto_currencies,
+                                                                          time_frames,
+                                                                          real_time_time_frames)
+        to_handle_time_frames = []
+        if self.time_frame is None:
+            self.logger.error("Missing self.time_frame value, impossible to initialize this evaluator.")
+        else:
+            ideal_time_frame = TimeFrames(self.time_frame)
+            to_handle_time_frame = ideal_time_frame
+            if to_handle_time_frame in real_time_time_frames:
+                to_handle_time_frame = TimeFrames(self.time_frame)
+            elif real_time_time_frames:
+                to_handle_time_frame = real_time_time_frames[0]
+            else:
+                to_handle_time_frame = time_frames[0]
+            if ideal_time_frame != to_handle_time_frame:
+                self.logger.warning(f"Missing {ideal_time_frame.name} time frame in available time frames, "
+                                    f"using {to_handle_time_frame.name} instead.")
+            to_handle_time_frames = [to_handle_time_frame]
+        # by default time frame registration only for the timeframe of this real-time evaluator
+        return currencies, symbols, to_handle_time_frames
```

Comment:
` return currencies, symbols, [to_handle_time_frame]`

Reply:
We need to keep the default value initialized above for error cases

Reply:
Yes you're right :+1: